helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -287,11 +267,18 @@ public void setInstanceEnabled(boolean enabled) {
     _record.setLongField(InstanceConfigProperty.HELIX_ENABLED_TIMESTAMP.name(),
         System.currentTimeMillis());
     if (enabled) {
-      _record.getSimpleFields().remove(InstanceConfigProperty.HELIX_DISABLED_REASON.toString());
-      _record.getSimpleFields().remove(InstanceConfigProperty.HELIX_DISABLED_TYPE.toString());
+      resetInstanceDisabledTypeAndReason();
     }
   }
 
+  /**
+   * Removes HELIX_DISABLED_REASON and HELIX_DISABLED_TYPE entry from simple field.
+   */
+  public void resetInstanceDisabledTypeAndReason() {
```

Comment:
Can we make this a private method?

Reply:
TFTR. This will be used in ZkHelixManager in later PR.

Reply:
Ok. Then let's leave it as is.